pkg/controller/clusterapi.go

```diff
@@ -41,3 +42,39 @@ func CapiClusterObjectKey(cluster *anywherev1.Cluster) client.ObjectKey {
 		Namespace: constants.EksaSystemNamespace,
 	}
 }
+
+// GetKubeadmControlPlane reads a cluster-api KubeadmControlPlane for an eks-a cluster using a kube client
+// If the KubeadmControlPlane is not found, the method returns (nil, nil).
+func GetKubeadmControlPlane(ctx context.Context, client client.Client, cluster *anywherev1.Cluster) (*controlplanev1.KubeadmControlPlane, error) {
+	kubeadmControlPlaneName := clusterapi.KubeadmControlPlaneName(cluster)
+
+	kubeadmControlPlane := &controlplanev1.KubeadmControlPlane{}
+	key := types.NamespacedName{Namespace: constants.EksaSystemNamespace, Name: kubeadmControlPlaneName}
+
+	err := client.Get(ctx, key, kubeadmControlPlane)
+	if apierrors.IsNotFound(err) {
+		return nil, nil
+	}
+
+	if err != nil {
+		return nil, err
+	}
+	return kubeadmControlPlane, nil
+}
+
+// GetMachineDeployment reads a cluster-api MachineDeployment of a given name using a kube client
+// If the MachineDeployment is not found, the method returns (nil, nil).
+func GetMachineDeployment(ctx context.Context, client client.Client, name string) (*clusterv1.MachineDeployment, error) {
```

Comment:
do we need a method for this? It's not encapsulating a particular logic to derive one object from another, you are just passing the name.
It's basically a wrapper around the client that just checks for not found error.

Reply:
I'll take this out. Calling it in `detectOperation()` will still require us to do a nil check, same as apierror.IsNotFound, so it won't save much trouble.